nuxeo-opencmis-impl/src/main/java/org/nuxeo/ecm/core/opencmis/impl/server/NuxeoContentStream.java

```diff
@@ -114,6 +119,28 @@ public static boolean isHeadRequest(HttpServletRequest request) {
         return request.getMethod().equals("HEAD");
     }
 
+    public static boolean hasWantDigestRequestHeader(HttpServletRequest request, String digestAlgorithm) {
+        if (request == null || digestAlgorithm == null) {
+            return false;
+        }
+        if (request instanceof HttpServletRequestWrapper) {
+            request = (HttpServletRequest) ((HttpServletRequestWrapper) request).getRequest();
```

Comment:
Is the unwrap needed? Doesn't `request.getHeaders` still work? If not, then please add a comment explaining why.